### ISSUE TITLE
Seeing odd caching error problem after upgrade from 1.15 to 1.16

### ISSUE THREAD

**Issue by jcope2013** [emberjs/data#2975]
versions

ember data 1.16.1
ember cli 0.2.1
ember 1.9.1

TLDR:

between the two releases, a record that was previously being treated as isEmpty is now being treated as isLoading

duplicating the same exact interaction in my app between versions, this happens, i would expect 1.15 to be right behavior

in 1.15 , if (get(record, 'isEmpty')) executes
in 1.16 else if (get(record, 'isLoading')) { executes

```
_findByRecord: function(record, preload) {
    var fetchedRecord;

    if (preload) {
      record._preloadData(preload);
    }

    if (get(record, 'isEmpty')) {
      fetchedRecord = this.scheduleFetch(record);
      //TODO double check about reloading
    } else if (get(record, 'isLoading')) {
      fetchedRecord = record._loadingPromise;
    }
```

I put in a fix to make sure my store gets completely unloaded when logging out and that solved it

explanation:

so the situation I am seeing is I have a session timeout on my server, I refresh the page, I try to get the session.currentUser in the navbar of my template, it sends out server request, I get a 401 unauthorized error back from my server then at about the same time i also get logged out from the app as the session has been timed out, everything is working fine and that is expected, so then I log back in, log in works fine then it gets into my index route 

the trouble starts here, calling this.container.lookup('store:main').find('user', userId) is actual erroring out with the cached 401 error from the previous response right before i got logged out and never even sending out a network request, i also tried this.container.lookup('store:main').fetchById('user', userId) with no success, in ember data 1.15 it is the opposite and that same code sends out a network request and works as i would expect it to work with no error happening, it doesn't use the previous 401 error response

so the gist of it is seems like something between the releases is now somehow actually caching errors when using store finder methods in certain scenarios, i diffed the two releases (1.15 and 1.16) and went through all the commits and don't see anything too obvious of what is causing this, 

maybe a pointer at what method i should be focusing on for someone with more expertise in ember data to help me try to diagnose this more would be a help also, if i had a guess it would somehow be related to the new snapshot api in some fashion?

down below is the code with comments showing the difference in behavior, I will also try to get together a jsbin showing this in some fashion

```
index.js route

 beforeModel: function(transition) {
    this._super(transition);
    var self = this;
    var applicationRoute = this.container.lookup('route:application');
    this.get('session.currentUser').then(function(user){
        self.transitionTo('providerDashboard'); // expected result on login after timeout, behavior in 1.15
    }, function(error) {
       Ember.onerror(error); // actual result on login after timeout, behavior in 1.16
    });
  }
});

simple auth initializer 

var CustomSession = SimpleAuthSession.extend({
  currentUser: function() {
    var userId = this.get('userId');
    if (!Ember.isEmpty(userId)) {
      // 1.15, works fine, sends out network request after login after timeout and refresh
     // 1.16 errors out, no network request even sent (uses previously cached 401 error from before i was logged out) after login after timeout and refresh
      return this.container.lookup('store:main').find('user', userId);
    }
  }.property('userId')
});

```
